src/main/java/net/foxes4life/foxclient/gui/BlockHud.java

```diff
@@ -0,0 +1,120 @@
+package net.foxes4life.foxclient.gui;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import net.foxes4life.foxclient.Main;
+import net.foxes4life.foxclient.configuration.FoxClientSetting;
+import net.foxes4life.foxclient.util.TextUtils;
+import net.foxes4life.foxclient.util.draw.Anchor;
+import net.foxes4life.foxclient.util.draw.AnchoredBounds;
+import net.foxes4life.foxclient.util.draw.DrawUtils;
+import net.foxes4life.foxclient.util.rendering.ItemRender;
+import net.minecraft.block.AirBlock;
+import net.minecraft.block.Block;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.client.gui.hud.InGameHud;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.client.network.ClientPlayerInteractionManager;
+import net.minecraft.client.render.RenderLayer;
+import net.minecraft.client.render.item.ItemRenderer;
+import net.minecraft.client.render.model.json.ModelTransformationMode;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.MutableText;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.RaycastContext;
+
+import java.awt.*;
+
+public class BlockHud {
+    private final MinecraftClient client;
+    private Block lastBlock;
+
+    private float y = 0;
+    private float width = 0;
+
+    public BlockHud(MinecraftClient client) {
+        this.client = client;
+    }
+
+    public void render(DrawContext context, ItemRender itemRender) {
+        Block block = getBlock();
+        boolean show = false;
+
+        if (block != null && !(block instanceof AirBlock)) {
+            lastBlock = block;
+            show = true;
+        }
+        else {
+            block = lastBlock;
+        }
+
+        // if the block is still null, we don't want to render anything
+        if (block == null) return;
+
+        boolean showAnimations = Main.config.get(FoxClientSetting.BlockHudAnimations, Boolean.class);
+
+        float delta = client.getTickDelta();
+
+        final int padding = 2;
+        final int border = 3;
+        final int height = 16;
+
+        final int minY = -height - padding * 2 - border - 2;
+        int goalY = show ? 0 : minY;
+
+        // transition y position
+        if (showAnimations) y = MathHelper.lerp(.6f * delta, y, goalY);
+        else y = goalY;
+
+        // don't even bother if its off screen
+        if (y == minY) return;
+
+        String translationKey = block.getTranslationKey();
```

Comment:
seems like an unused variable assignment, could be directly put into the translatable method call next line